### DISCUSSION TITLE
Is it possible to add custom options to template in frontmatter configuration?

### DISCUSSION THREAD

**Discussion by LuHugo** [withastro/starlight#1112] [Feature Requests]
### What version of `starlight` are you using?

0.13.0

### What is your idea?

In the frontmatter configuration `template`, you can add the `custom` option to eliminate the restrictions of the default page style.
```js
---
title: widescreen page
template: custom
---

```
When the template is set to custom, there are no built-in components in the page except for the navigation bar at the top. It is empty and has full width.

In this way, the mdx page can not be limited by the internal width and remove unnecessary built-in components.

### Why is this feature necessary?

This method is useful when the mdx page needs to be widescreen or when you need to customize the hero to have more display forms.

### Participation

- [ ] I am willing to submit a pull request for this feature.

**Comment by delucis** [comment]
We could definitely consider this. I think someone else suggested a `template: empty` for a similar use case.

**Reply by mattjennings** [reply]
I would definitely appreciate this as an alternative to splash for the landing page. It feels a bit awkward on the hero when you don't have a logo to use, and I'm not creative enough to come up with one just to fill the space 😅

**Reply by sanscontext** [reply]
This would also help me with clearing out the layouts so I can insert my Rapidoc nonsense.

**Comment by clemwo** [comment]
Hi, this sounds interesting.
I'm not entirely sure whether that helps with my use case so I'd love to hear some opinions.
I want to build my home page using Astro Starlight and would love to integrate some tools on my page such as a quick toUppercase textbox with a button and other stuff like this.

For this I basically want to create astro pages that still have the header, sidebar and so on but other than that provide no styling.

Would that be the usecase for template empty?

**Reply by delucis** [reply]
I think the “empty” template proposal above didn’t include the sidebar, but good to know there’s interest in that too.

If you were happy to still have Starlight’s page title on these pages, you could disable just the table of contents, which would get you basically that: header, sidebar, title and then the rest up to you.
